lib/report-generator.js

```diff
@@ -0,0 +1,95 @@
+var fs = require('fs-extra'),
+    path = require('path');
+
+/**
+ * The directory of the visual report
+ *
+ * @type {String}
+ * @const
+ * @default
+ */
+var DIRECTORY_PATH = 'visual-report';
+
+/**
+ * The directory of the static files
+ *
+ * @type {String}
+ * @const
+ * @default
+ */
+var STATIC_DIRECTORY = 'statics';
+
+/**
+ * The path of the data file
+ *
+ * @type {String}
+ * @const
+ * @default
+ */
+var DATA_PATH = 'data.js';
+
+/**
+ * The root directory
+ *
+ * @type {String}
+ */
+var ROOT_DIRECTORY;
+
+
+/**
+ * Creates the files for the visual report
+ *
+ * @param {Array.<Suite>} data - The generated data from Mocha's runner
+ * @param {Object} options - Options object
+ * @param {String} options.rootDirectory - The directory of the visual report
+ * @param {generateReportCb} done - Cb to notify when the report is created
+ */
+module.exports = function(data, options, done) {
+  if (data === undefined) {
+    throw new Error('No received data to create the "data.js" file');
+  }
+
+  if (typeof options === 'function') {
```

Comment:
When there is no `options object`, it means that the `options obj is the callback` and I set it appropriately.

I have thought maybe in some future cases we won't provide an options object, i.e this part of the reporter can be used also for other reporters (ex. Jasmine reporter) and maybe you cannot pass any reporter options and so you cannot change the `rootDirectory` so we will not pass an options object. 

I know we can pass an empty object, `{}`. @andrei-picus-hs do you want to remove this ? And make the assumption that there will always be an `options obj`?